File: apps/server/src/university/faculties/dtos.rs
use serde::{Deserialize, Serialize};
use validator::Validate;

#[derive(Debug, Serialize, Deserialize, Validate)]
pub struct CreateFacultyDto {
    #[validate(length(
        min = 1,
        max = 255,
        message = "El nombre de la facultad debe tener entre 1 y 255 caracteres"
    ))]
    pub name: String,
}

#[derive(Debug, Serialize, Deserialize, Validate, Default)]
pub struct GetFacultiesQuery {
    #[validate(length(
        min = 1,
        max = 255,
        message = "El nombre de la facultad debe tener entre 1 y 255 caracteres"
    ))]
    pub name: Option<String>,
}

File: apps/client/src/lib/university/faculties/dtos.ts
import * as v from "valibot"

export interface FacultyDTO {
	id: string
	name: string
}

export interface GetFacultiesParams {
	name?: string
}

export const createFacultySchema = v.object({
	name: v.pipe(
		v.string(),
		v.minLength(1, "El nombre debe tener entre 1 y 255 caracteres"),
		v.maxLength(255, "El nombre debe tener entre 1 y 255 caracteres"),
	),
})

export type CreateFacultyDto = v.InferInput<typeof createFacultySchema>

export const createFacultyDTOInitialInput = {
	name: "",
}
